### PR TITLE
feat: add more southern names and extend last names using basic names

### PR DESCRIPTION
**Pull request by LordMidas**
This adds more variety for southern names:
- Add 59 new names.
- Extend the vanilla "southern last names" list by prefixing `Ibn` and `Al-` to all the first names.

### CODE REVIEW

**Comment by Uberbagel** [comment]
> Extend the vanilla "southern last names" list by prefixing Ibn and Al- to all the first names.

Can you explain what this means in practical terms?

**Comment by LordMidas** [comment]
> > Extend the vanilla "southern last names" list by prefixing Ibn and Al- to all the first names.
> 
> Can you explain what this means in practical terms?

`Ibn` means "son of` and `Al-` means "the" in Arabic. Vanilla has two lists `SouthernNames` and `SouthernNamesLast`. It generates characters by using a name from `SouthernNames` and then sometimes it affixes a last name from `SouthernNamesLast`. These last names in vanilla are composed of formations such as `Ibn Sahr` or `Al-Faris`. I think there is no reason why can't expand that setup by just prefixing `Ibn` and `Al-` to all the first names. This results in a lot more possible valid last names.

**Comment by Uberbagel** [comment]
So now we lose Ibn Sahr and Al-Faris and only get Ibn Al-???

**Comment by LordMidas** [comment]
> So now we lose Ibn Sahr and Al-Faris and only get Ibn Al-???

No. We extend the last names list by prefixing Ibn and Al- to all the first names. So if a first name was "Adil" for example then both "Ibn Adil" and "Al-Adil" are now additional valid last names to roll from.

File: mod_reforged/hooks/config/character_names.nut [hunk 1]
```diff
@@ -242,3 +242,71 @@ foreach(characterName in ::Const.Strings.CharacterNames)
...
+	"Zuhayr",
+	"Zahid",
+	"Zubayr"
+]);
+
+foreach (name in ::Const.Strings.SouthernNames)
+{
+	::Const.Strings.SouthernNamesLast.push("Ibn " + name);
+	::Const.Strings.SouthernNamesLast.push("Al-" + name);
+}
```
Comment (Darxo) [inline mod_reforged/hooks/config/character_names.nut:312]: The name `Wasi` appears in `SouthernNames` and also in `SouthernNamesLast`. There might be more name overlaps.
Furthermore as described above, two names appear twice in the list bove.

You should remove duplicates after generating this name list, like we do above with the knight names

**Review by Darxo — CHANGES_REQUESTED** [review]

**Comment by Uberbagel** [comment]
Sounds good then

**Review by Uberbagel — APPROVED** [review]

**Review by Darxo — APPROVED** [review]

**Review by Darxo — APPROVED** [review]

**Review by Uberbagel — APPROVED** [review]

**Pull request merged**